### PR TITLE
feat: Add `CheckOption` on Omnichannel's departments multi selects

### PR DESCRIPTION
**Pull request by csuadev**
Change the Option component in favor of CheckOption component to improve the UX to the end user and allow them to see already selected elements when using MultiSelects avoiding confusion when there are multiple items. 

## Proposed changes (including videos or screenshots)
Before: 
image

image

After: 
image

image

## Issue(s)
[https://rocketchat.atlassian.net/browse/SUP-400](https://rocketchat.atlassian.net/browse/SUP-400)

### CODE REVIEW

**Comment by dougfabris** [comment]
what is a `CheckPoint` component man?

**Review by MarcosSpessatto — DISMISSED** [review]

**Review by jessicaschelly — DISMISSED** [review]

**Review by MartinSchoeler — DISMISSED** [review]

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]
Can you check the latest build, it seems there is a flaky test related to omnichannel units, let's check if its related to any of the changes done to tests in this PR

Screenshot 2024-04-29 at 11 02 56

**Review by jessicaschelly — DISMISSED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts [hunk 1]
```diff
@@ -138,29 +138,77 @@ test.describe('OC - Manage Units', () => {
...
 			});
 
-			await test.step('expect to have been deleted', async () => {
-				await expect(poOmnichannelUnits.inputSearch).toBeVisible();
-				await poOmnichannelUnits.inputSearch.clear();
-				await expect(page.locator('h3 >> text="No units yet"')).toBeVisible();
+			return unit;
+		});
+
+		const department2 = await createDepartment(api);
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:164]: Can we delete this department afterAll?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts [hunk 2]
```diff
@@ -103,24 +105,25 @@ test.describe('OC - Manage Units', () => {
...
 				departments: [{ departmentId: department.data._id }],
 			});
 
-			return unit;
+			return newUnit;
 		});
 
-		await page.goto('/omnichannel');
-		await poOmnichannelUnits.sidenav.linkUnits.click();
+		await page.reload();
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:123]: why reloading this page?

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts [hunk 3]
```diff
@@ -103,24 +105,25 @@ test.describe('OC - Manage Units', () => {
...
 		
 		await test.step('expect to edit unit', async () => {
+			await poOmnichannelUnits.inputSearch.isVisible()
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:126]: there's no need to await this input to be visible, the locator already does it

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts [hunk 4]
```diff
@@ -138,29 +141,75 @@ test.describe('OC - Manage Units', () => {
...
 			await expect(poOmnichannelUnits.contextualBar).toBeVisible();
 
-			await test.step('expect confirm delete unit', async () => {
-				await test.step('expect to be able to cancel delete', async () => {
-					await poOmnichannelUnits.btnDelete.click();
-					await expect(poOmnichannelUnits.confirmDeleteModal).toBeVisible();
-					await poOmnichannelUnits.btnCancelDeleteModal.click();
-					await expect(poOmnichannelUnits.confirmDeleteModal).not.toBeVisible();
-				});
+			await test.step('expect to confirm delete', async () => {
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:144]: these amount of stuff is unnecessary, left just the test on the list of units after deleting it

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts [hunk 5]
```diff
@@ -138,29 +141,75 @@ test.describe('OC - Manage Units', () => {
...
 			});
 
-			await test.step('expect to have been deleted', async () => {
-				await expect(poOmnichannelUnits.inputSearch).toBeVisible();
-				await poOmnichannelUnits.inputSearch.clear();
-				await expect(page.locator('h3 >> text="No units yet"')).toBeVisible();
+			return unit;
+		});
+
+		await page.reload();
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:167]: why reloading it?

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Review by dougfabris — DISMISSED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by KevLehman — APPROVED** [review]

**Review by jessicaschelly — APPROVED** [review]

**Pull request merged**